frame/wormhole/backing/parachain/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+	pub type RemoteMappingTokenFactoryAccount<T: Config> =
+		StorageValue<_, AccountId<T>, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub secure_limited_period: BlockNumberFor<T>,
+		pub secure_limited_ring_amount: RingBalance<T>,
+		pub remote_mapping_token_factory_account: AccountId<T>,
+	}
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				secure_limited_period: Zero::zero(),
+				secure_limited_ring_amount: Zero::zero(),
+				remote_mapping_token_factory_account: Default::default(),
+			}
+		}
+	}
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			<SecureLimitedPeriod<T>>::put(self.secure_limited_period);
+			<SecureLimitedRingAmount<T>>::put((
+				<RingBalance<T>>::zero(),
+				self.secure_limited_ring_amount,
+			));
+			<RemoteMappingTokenFactoryAccount<T>>::put(
+				self.remote_mapping_token_factory_account.clone(),
+			);
+		}
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(now: BlockNumberFor<T>) -> Weight {
+			let secure_limited_period = <SecureLimitedPeriod<T>>::get();
+
+			if !secure_limited_period.is_zero() && (now % secure_limited_period).is_zero() {
+				<SecureLimitedRingAmount<T>>::mutate(|(used, _)| *used = Zero::zero());
+
+				T::DbWeight::get().reads_writes(2, 1)
+			} else {
+				T::DbWeight::get().reads(1)
+			}
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Lock token in this chain and cross transfer to the target chain
+		///
+		/// Target is the id of the target chain defined in s2s_chain pallet
+		#[pallet::weight(<T as Config>::WeightInfo::lock_and_remote_issue())]
+		#[transactional]
+		pub fn lock_and_remote_issue(
+			origin: OriginFor<T>,
+			spec_version: u32,
+			weight: u64,
+			#[pallet::compact] value: RingBalance<T>,
+			#[pallet::compact] fee: RingBalance<T>,
+			recipient: AccountId<T>,
+		) -> DispatchResultWithPostInfo {
+			let user = ensure_signed(origin)?;
+
+			// Make sure the locked value is less than the max lock limited
+			ensure!(
+				value < T::MaxLockRingAmountPerTx::get() && !value.is_zero(),
+				<Error<T>>::RingLockLimited
+			);
+			// Make sure the user's balance is enough to lock
+			let total_cost = value + fee;
+			ensure!(
+				T::RingCurrency::free_balance(&user) > total_cost,
+				<Error<T>>::InsufficientBalance
+			);
+
+			// this pallet account as the submitter of the remote message
+			// we need to transfer fee from user to this account to pay the bridge fee
+			T::RingCurrency::transfer(&user, &Self::pallet_account_id(), total_cost, KeepAlive)?;
+
+			let payload = T::OutboundPayloadCreator::create(
+				CallOrigin::SourceAccount(Self::pallet_account_id()),
+				spec_version,
+				weight,
+				IssuingCall::<T>::ParachainIssuingPalletIssueFromRemote(value, recipient.clone()),
```

Comment:
Be care about the decimal conversation here.

Reply:
Handled here:

https://github.com/darwinia-network/darwinia-parachain/pull/85/files#diff-cd4bcbd2cbdeec4c0771723670222324251a14c1d277112929935c99d212e375R216-R218